core/src/main/java/org/apache/accumulo/core/client/admin/TabletInformation.java

```diff
@@ -0,0 +1,159 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.client.admin;
+
+import java.util.Objects;
+
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.util.NumUtil;
+
+public class TabletInformation {
+
+  private final String tableName;
+  private final String endRow;
+  private final String prevEndRow;
+  private final int numFiles;
+  private final int numWalLogs;
+  private final long numEntries;
+  private final long size;
+  private final String status;
+  private final String location;
+  private final String dir;
+  private final TableId tableId;
+  private final TabletHostingGoal goal;
+
+  public TabletInformation(String tableName, TableId tableId, String endRow, String prevEndRow,
+      int numFiles, int numWalLogs, long numEntries, long size, String status, String location,
+      String dir, TabletHostingGoal goal) {
+    this.tableName = tableName;
+    this.tableId = tableId;
+    this.endRow = endRow;
+    this.prevEndRow = prevEndRow;
+    this.numFiles = numFiles;
+    this.numWalLogs = numWalLogs;
+    this.numEntries = numEntries;
+    this.size = size;
+    this.status = status;
+    this.location = location;
+    this.dir = dir;
+    this.goal = goal;
+  }
+
+  public String getEndRow() {
+    return Objects.requireNonNullElse(this.endRow, "+INF");
+  }
+
+  public String getStartRow() {
+    return Objects.requireNonNullElse(this.prevEndRow, "-INF");
+  }
+
+  public String getTableId() {
+    return tableId.canonical();
+  }
+
+  public String getTablet() {
+    return getStartRow() + " " + getEndRow();
+  }
+
+  public String getTableName() {
+    return this.tableName;
+  }
+
+  public int getNumFiles() {
+    return this.numFiles;
+  }
+
+  public int getNumWalLogs() {
+    return this.numWalLogs;
+  }
+
+  public long getNumEntries() {
+    return this.numEntries;
+  }
+
+  public String getNumEntries(final boolean humanReadable) {
+    if (humanReadable) {
+      return NumUtil.bigNumberForQuantity(numEntries);
+    }
+    return String.format("%,d", numEntries);
+  }
+
+  public long getSize() {
+    return this.size;
+  }
+
+  public String getSize(final boolean humanReadable) {
+    if (humanReadable) {
+      return NumUtil.bigNumberForSize(size);
+    }
+    return String.format("%,d", size);
+  }
+
+  public String getStatus() {
+    return this.status;
+  }
+
+  public String getLocation() {
```

Comment:
Since a tablet may not have a location, could have an optional return type to make this clear.

```suggestion
  public Optional<String> getLocation() {
```

Also as follow on issue, this could possible use the new type introduced in #3626